diff --git a/dist/esm/index.js b/dist/esm/index.js
index 659542b7d51f2c3d988409f18b4a1fd96831d2a5..f01228b12bc0fb4167dc8deb8655000060d5f3e1 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -462,10 +462,14 @@ function GarfishRouter(_args) {
             return active2(appInfo, rootPath);
           appInfo.rootPath = rootPath;
           const currentApp = activeApp = createKey();
+          let entry=appInfo.entry;
+          if(typeof entry==='function'){
+            entry=entry();
+          }
           const app = await Garfish.loadApp(appInfo.name, {
             cache,
             basename: rootPath,
-            entry: appInfo.entry,
+            entry,
             domGetter: appInfo.domGetter
           });
           if (app) {