diff --git a/dist/client/index.js b/dist/client/index.js
index 2848e4da7a6b0f5fdea209ce999dec5c47206f5f..628200538802f0daa595ee0e0e3052e5a1cb55ec 100644
--- a/dist/client/index.js
+++ b/dist/client/index.js
@@ -296,6 +296,7 @@ function Head({ callback  }) {
     return null;
 }
 let reactRoot = null;
+Object.defineProperty(exports,'reactRoot',{ get(){ return reactRoot; }});
 // On initial render a hydrate should always happen
 let shouldHydrate = true;
 function clearMarks() {